src/Shared/FileMatcher.cs

```diff
@@ -372,7 +412,7 @@ GetFileSystemEntries getFileSystemEntries
                     else
                     {
                         // getFileSystemEntries(...) returns an empty array if longPath doesn't exist.
-                        string[] entries = getFileSystemEntries(FileSystemEntity.FilesAndDirectories, longPath, parts[i], null, false);
+                        string[] entries = getFileSystemEntries(FileSystemEntity.FilesAndDirectories, longPath, parts[i], null, false).ToArray();
```

Comment:
This can throw.

Reply:
[Fixed](https://github.com/maca88/msbuild/blob/b5574289426c99e9567a43fe17d4731e1f4f5380/src/Shared/FileMatcher.cs#L221)